lib.rs

```diff
@@ -148,15 +148,86 @@ fn error(e: &str) -> TokenStream {
     TokenStream::from_str(&format!("::core::compile_error!{{\"{}\"}}", e.escape_default())).unwrap()
 }
 
+fn compile_error(msg: &str, tt: Option<TokenTree>) -> TokenStream {
```

Comment:
there was a `error` function just before, that does more or less the same.
(The main difference being the Span, but this could be added)

Reply:
I know, but the span is crucial for usability. Run `cargo test -- --nocapture` to see the nice error messages produced by the doc-tests marked as `compile_fail`. Without the span indication the error messages would be confusing. For instance `` expected `=` `` would not mark the position where `=` is expected.

Reply:
yes indeed. i was just wondering if it was still possible to add the span when doing from_str.  but if not, then never mind.